### PR TITLE
Implement extended CRUD operations

### PR DESCRIPTION
**Pull request by owenwexler**

### CODE REVIEW

File: DB/ReviewsController.js [hunk 1]
```diff
@@ -7,14 +7,47 @@ const getAll = (callback) => {
...
+    if (err) {
+      callback(err, null);
+    } else {
+      callback(null, result);
+    }
+  });
+}
+
+const updateById = (id, updateObj, callback) => {
+  Reviews.findOneAndUpdate({id : `${id}`}, updateObj, (err, result) => {
```
Comment (luke-eastman) [inline DB/ReviewsController.js:40]: I don't know that template literals are necessary here. I would think you could just use the variable without them.

File: Server/server.js [hunk 2]
```diff
@@ -39,15 +39,12 @@ app.get('/api/reviews', (req, res) => {
...
-  //   res.send(results);
-  // })
 })
 
 app.get('/:id', (req, res) => {
   ReviewsController.getById(req.params.id, (err, result) => {
     if (err) {
-      console.log('ERROR')
+      console.log('ERROR');
       throw new Error();
```
Comment (luke-eastman) [inline Server/server.js]: I know this isn't your part of the code, but is just throwing a new error what you want to do here?

File: Server/server.js [hunk 3]
```diff
@@ -58,10 +55,45 @@ app.get('/:id', (req, res) => {
...
+    } else {
+      res.send(result);
+    }
+  })
+});
+
+app.put('/api/reviews/:id', (req, res) => {
+  ReviewsController.updateById(req.params.id, req.body, (err, result) => {
+    if (err) {
+      res.send(err);
```
Comment (luke-eastman) [inline Server/server.js]: I don't actually know the proper way to handle errors but I think you may want to be consistent across all your endpoints, one way or another.

**Review by luke-eastman — APPROVED** [review]
Looks good! Just left a couple notes about error handling.

File: Server/server.js (hunk 2 above)
Comment (owenwexler) [reply Server/server.js]: Very good point.  I'm still trying to decide whether to keep this type of error handling.  It's going to depend on what the front end is doing.  Will take a look after prezzies.

**Review by j-tirelli — APPROVED** [review]
This all looks good Owen!

If you wanted to be fully explicit you could also list your status codes for get as 200 (I think thats done by default) and your puts, deletes and posts as 204 (Unless you post returns a response body). [Source](https://restfulapi.net/http-status-codes/)

**Comment by owenwexler** [comment]
Good call... that's being done tomorrow.

**Pull request merged**